sympy/polys/rings.py

```diff
@@ -2216,6 +2218,33 @@ def _gcd_QQ(self, g):
 
         return h, cff, cfg
 
+    def _gcd_I(self, g):
```

Comment:
This should not be called `_gcd_I` because it is not specific to `I` even if that is the most common case that arises in the tests.